### PR TITLE
Adventure

### PR DESCRIPTION
**Pull request by TrueFuFLeaderG**
ONLY ADVENTURE CHANGES
i changed the pathing for the roads
added WaveFunctionCollapse  to generate terrain on the overworld
added collision objects for enemies and the player on the overworld

does someone know why the commits got so messed up, if i pull from main/master everything is up to date and it does run on my side, but a lot of changes that i did not do are in this PR.

### CODE REVIEW

File: forge-gui-mobile/src/forge/adventure/data/BiomeStructureData.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package forge.adventure.data;
+
+import java.awt.image.BufferedImage;
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/data/BiomeStructureData.java:3]: Does this class is only accessed via the desktop adventure editor? mobile version (android) don't have BufferedImage

File: forge-gui-mobile/src/forge/adventure/stage/GameHUD.java [hunk 2]
```diff
@@ -230,19 +238,22 @@ public void draw() {
...
+            miniMapToolTipTexture.dispose();
+        if(miniMapToolTipPixmap!=null)
+            miniMapToolTipPixmap.dispose();
+        miniMapToolTipPixmap=new Pixmap((int) (miniMap.getWidth()*3), (int) (miniMap.getHeight()*3), Pixmap.Format.RGBA8888);
+        miniMapToolTipPixmap.drawPixmap(WorldSave.getCurrentSave().getWorld().getBiomeImage(),0,0,WorldSave.getCurrentSave().getWorld().getBiomeImage().getWidth(),WorldSave.getCurrentSave().getWorld().getBiomeImage().getHeight(),0,0,miniMapToolTipPixmap.getWidth(),miniMapToolTipPixmap.getHeight());
+        miniMapToolTipTexture=new Texture(miniMapToolTipPixmap);
         miniMap.setDrawable(new TextureRegionDrawable(miniMapTexture));
+        miniMapTooltip.setDrawable(new TextureRegionDrawable(miniMapToolTipTexture));
         avatar.setDrawable(new TextureRegionDrawable(Current.player().avatar()));
-        Deck deck = AdventurePlayer.current().getSelectedDeck();
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/stage/GameHUD.java:240]: I added this as an Indicator to player that they have empty or incomplete  deck selected so they don't battle with deck full of wastes accidentally incase the selected deck is empty (via accident or for some unknown reason). Do you have an alternative replacement for this or any hints to the player?

File: forge-gui-mobile/src/forge/adventure/world/BiomeStructure.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
+package forge.adventure.world;
+
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.github.sjcasey21.wavefunctioncollapse.OverlappingModel;
+import forge.adventure.data.BiomeStructureData;
+import forge.adventure.util.Config;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/world/BiomeStructure.java:9]: Does this class is only accessed via the desktop adventure editor? mobile version (android) don't have BufferedImage and ImageIO is not part of mobile platform on android.

File: forge-gui-mobile/src/forge/adventure/data/BiomeStructureData.java (hunk 1 above)
Comment (TrueFuFLeaderG) [reply forge-gui-mobile/src/forge/adventure/data/BiomeStructureData.java:3]: sadly it is used by the api of the WaveLengthCollapse algo.
Then I have to copy yoink it and copy it to the project to rewrite it...

File: forge-gui-mobile/src/forge/adventure/stage/GameHUD.java (hunk 2 above)
Comment (TrueFuFLeaderG) [reply forge-gui-mobile/src/forge/adventure/stage/GameHUD.java:240]: I thought I merged it, but as you can see the commits are quite a mess right now apperently it got removed again, I will check it later

**Pull request merged**